internal/state/hooks.go

```diff
@@ -9,3 +9,9 @@ func (mh ModuleChangeHooks) notifyModuleChange(oldMod, newMod *Module) {
 		h(oldMod, newMod)
 	}
 }
+
+func (mh ModuleChangeHooks) notifyTokenRefresh(oldMod, newMod *Module) {
```

Comment:
Is there a reason we can't just reuse the existing hook which notifies of a module change and just filter the changes to avoid refreshing e.g. when schema doesn't change?

Reply:
No reason other than I was stumbling through remembering how all this worked and piecing it together as I went. Will modify.

I am also thinking to review where/when we notify in light of how many times this is firing right now. I worry about cpu usage in larger projects. More testing is needed here.

Also am fighting some build errors, but haven't looked into that yet.

Reply:
I wouldn't expect the hook firing itself and comparison of data within the hook (which then results to returning early and _not_ doing anything else) to be expensive in the grand scheme of things but I haven't done any extensive testing to confirm that theory admittedly.

Also I expect that the times when we want to fire the other two hooks (diagnostics and telemetry) will overlap a lot with this one anyway, so I'm not sure we'd gain all that much from spreading the logic through the `ModuleStore` methods.

Reply:
Awesome. Kept the hooks in the current places.